src/d3a/models/strategy/external_strategies/storage.py

```diff
@@ -254,6 +258,7 @@ def _bid_impl(self, arguments, bid_response_channel):
                 buyer_origin=arguments["buyer_origin"]
             )
             self.state.offered_buy_kWh[self.market.time_slot] += bid.energy
+            self.state.energy_to_buy_dict[self.market.time_slot] -= bid.energy
```

Comment:
Same here, you should use the `get_posted_energy` and `clamp_energy_to_buy` methods.